cmd/cleanup-controller/handlers/cleanup/handlers.go

```diff
@@ -55,19 +56,48 @@ func (h *handlers) lookupPolicy(namespace, name string) (kyvernov1alpha1.Cleanup
 func (h *handlers) executePolicy(ctx context.Context, logger logr.Logger, policy kyvernov1alpha1.CleanupPolicyInterface) error {
 	spec := policy.GetSpec()
 	kinds := sets.NewString(spec.MatchResources.GetKinds()...)
+	var errs []error
 	for kind := range kinds {
 		logger := logger.WithValues("kind", kind)
 		logger.Info("processing...")
 		list, err := h.client.ListResource(ctx, "", kind, policy.GetNamespace(), nil)
 		if err != nil {
-			return err
-		}
-		for i := range list.Items {
-			if !controllerutils.IsManagedByKyverno(&list.Items[i]) {
-				logger := logger.WithValues("name", list.Items[i].GetName(), "namespace", list.Items[i].GetNamespace())
-				logger.Info("item...")
+			logger.Error(err, "failed to list resources")
+			errs = append(errs, err)
+		} else {
+			for i := range list.Items {
+				if !controllerutils.IsManagedByKyverno(&list.Items[i]) {
+					logger := logger.WithValues("name", list.Items[i].GetName(), "namespace", list.Items[i].GetNamespace())
+					// match resource with match/exclude clause
+					matched := checkMatchesResources(list.Items[i], spec.MatchResources, nil, nil)
+					if matched != nil {
+						logger.Info("resource/match didn't match", "result", matched)
```

Comment:
Can we set these to debug level (L74, L83, L87) and have the final result logged at default level 2?

Reply:
Sure